openhands/llm/async_llm.py

```diff
@@ -39,14 +39,8 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
 
         async_completion_unwrapped = self._async_completion
 
-        @self.retry_decorator(
-            num_retries=self.config.num_retries,
-            retry_exceptions=LLM_RETRY_EXCEPTIONS,
-            retry_min_wait=self.config.retry_min_wait,
-            retry_max_wait=self.config.retry_max_wait,
-            retry_multiplier=self.config.retry_multiplier,
-        )
-        async def async_completion_wrapper(*args: Any, **kwargs: Any) -> Any:
+        # Define the wrapper function
+        async def async_completion_impl(*args: Any, **kwargs: Any) -> Any:
```

Comment:
Is there a way to avoid moving this decorator? It seems simpler to have it as a decorator.